library/ssl_tls.c

```diff
@@ -7098,12 +7098,12 @@ void mbedtls_ssl_handshake_free( mbedtls_ssl_context *ssl )
 #endif
 #if defined(MBEDTLS_ECDH_C)
 
-#if !defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2_OR_EARLIER)
```

Comment:
Such `#if TLS 1.2 ... #else ... #endif` branches are not yet suitable for dual 1.2 and 1.3 builds: We would rather want independent `#if TLS 1.2 ... #endif #if TLS 1.3 .. #endif` blocks. However, I understand that this requires more work - in the snippet at hand, for example, it wouldn't work at the moment because `ecdh_ctx` is overloaded.